docs/reference/protocol-spec.md

```diff
@@ -268,25 +313,278 @@ Null element:
     $5\r\n
     world\r\n
 
-The second element is a Null. The client library should return something
-like this:
+Above, the second element is a Null.
+The client library should return to its caller something like this:
 
     ["hello",nil,"world"]
 
-Note that this is not an exception to what was said in the previous sections, but 
-an example to further specify the protocol.
+<a name="null-reply"></a>
+
+### Nulls
+The Null data type represents non-existent values.
+
+Nulls' encoding is the underscore (`_`) character, followed by the CRLF terminator (`\r\n`).
+Here's Null's raw RESP encoding:
+
+    _\r\n
+
+{{% alert title="Null Bulk String, Null Arrays and Nulls" color="info" %}}
+Due to historical reasons, RESP2 features two specially-crafted values for representing null values of Bulk Strings and Arrays.
+This duality has always been a redundancy that added zero semantical value to the protocol itself.
+
+The Null type, introduced in RESP3, aims to fix this wrong.
+{{% /alert %}}}}
+
+<a name="boolean-reply">
+
+### Booleans
+RESP Booleans encode true and false values in the following ways:
```

Comment:
why not use the same format as other types?

Booleans are encoded as follows:

    #<boolean>\r\n

* The "number sight" character (`#`) as the first byte.
* `t` or `f` (`true` or `false`).
* The CRLF terminator.

Reply:
I did it to break out of the routine. I guess that's unacceptable. Fixed.